pkgs/by-name/ma/magento-cloud/package.nix

```diff
@@ -0,0 +1,59 @@
+{
+  makeBinaryWrapper,
+  stdenvNoCC,
+  php,
+  lib,
+  fetchurl,
+}:
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "magento-cloud";
+  version = "1.46.1";
+
+  src = fetchurl {
+    url = "https://accounts.magento.cloud/sites/default/files/magento-cloud-v${finalAttrs.version}.phar";
+    hash = "sha256-QrrD5pz6Juov1u3QYcuLr6aEKe/4DX5wFKs+hp6KjJ8=";
+  };
+
+  dontUnpack = true;
+
+  nativeBuildInputs = [ makeBinaryWrapper ];
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+    install -D ${finalAttrs.src} $out/libexec/magento-cloud/magento-cloud.phar
+    makeWrapper ${lib.getExe php} $out/bin/magento-cloud \
+      --add-flags "$out/libexec/magento-cloud/magento-cloud.phar"
+
+    runHook postInstall
+  '';
+
+  doInstallCheck = true;
+
+  # Workaround for "Fatal error: Uncaught RuntimeException: Could not determine home directory"
+  # Don't use versionCheckHook for this package. It cannot inject environment variables.
+  installCheckPhase = ''
+    if [[ "$(HOME="$(mktemp -d)" "$out/bin/magento-cloud" --version)" =~ "${finalAttrs.version}" ]]; then
```

Comment:
You can override the `HOME` env. variable in a `preCheck` attribute. Look in `nixpkgs`, it's a common practice.

Reply:
If my MR https://github.com/NixOS/nixpkgs/pull/370869 had been merged, this would have been as easy as adding a new dependency in `nativeCheckInputs` !

Reply:
I still won't be able to use `versionCheckHook` because environment variables are not passed to it: https://github.com/NixOS/nixpkgs/blob/269d8608343bfedf649232491a3ac3f73b359a15/doc/hooks/versionCheckHook.section.md#versioncheckhook-versioncheckhook

> It does so in a clean environment (using env --ignore-environment), and it checks for the ${version} string in both the stdout and the stderr of the command. It will report to you in the build log the output it received and it will fail the build if it failed to find ${version}.

Reply:
I could move adding home to preCheck as you mentioned:

```
  preInstallCheck = ''
    export HOME=$TMPDIR
  '';

  # Workaround for "Fatal error: Uncaught RuntimeException: Could not determine home directory"
  # Don't use versionCheckHook for this package. It cannot inject environment variables.
  installCheckPhase = ''
    runHook preInstallCheck

    if [[ "$("$out/bin/magento-cloud" --version)" =~ "${finalAttrs.version}" ]]; then
      echo '${finalAttrs.pname} version check passed.'
    else
      echo '${finalAttrs.pname} version check failed.'
      return 1
    fi

    runHook postInstallCheck
  '';
```
If it makes it cleaner.

Reply:
I think @GaetanLepage might be helpful here, he's more knowledgeable than me when it comes to version checks.